easy_my_coop/models/partner.py

```diff
@@ -1,170 +1,178 @@
+                                    compute="_compute_coop_candidate",
+                                    store=True,
+                                    readonly=True)
+    old_member = fields.Boolean(string='Old cooperator',
+                                help="Check this box if this cooperator is"
+                                " no more an effective member.")
+    # todo use oca partner_contact_gender
+    gender = fields.Selection([('male', 'Male'),
+                               ('female', 'Female'),
+                               ('other', 'Other')],
+                              string='Gender')
+    share_ids = fields.One2many('share.line',
+                                'partner_id',
+                                string='Share Lines')
+    cooperator_register_number = fields.Integer(string='Cooperator Number')
+    number_of_share = fields.Integer(compute="_compute_share_info",
+                                     multi='share',
+                                     string='Number of share',
+                                     readonly=True)
+    total_value = fields.Float(compute="_compute_share_info",
+                               multi='share',
+                               string='Total value of shares',
+                               readonly=True)
+    company_register_number = fields.Char(string='Company Register Number')
+    cooperator_type = fields.Selection(selection='_get_share_type',
+                                       compute=_compute_cooperator_type,
+                                       string='Cooperator Type',
+                                       store=True)
+    effective_date = fields.Date(sting="Effective Date",
+                                 compute=_compute_effective_date,
+                                 store=True)
+    representative = fields.Boolean(string="Legal Representative")
+    subscription_request_ids = fields.One2many('subscription.request',
+                                               'partner_id',
+                                               string="Subscription request")
+    legal_form = fields.Selection([('', '')],
+                                  string="Legal form")
+    data_policy_approved = fields.Boolean(string="Approved Data Policy")
+    internal_rules_approved = fields.Boolean(string="Approved Internal Rules")
+
+    @api.multi
+    @api.depends('subscription_request_ids.state')
+    def _compute_coop_candidate(self):
+        for partner in self:
+            if partner.member:
+                is_candidate = False
+            else:
+                if len(partner.subscription_request_ids.filtered(lambda record: record.state == 'done')) > 0:
+                    is_candidate = True
+                else:
+                    is_candidate = False
+
+            partner.coop_candidate = is_candidate
+
+    def has_representative(self):
+        if self.child_ids.filtered('representative'):
+            return True
+        return False
```

Comment:
```python
def has_representative(self):
    return bool(self. get_representative)
```

Shouldn't these functions be decorated with `@api.multi`? Or at least prefixed with `_` to mark it as private?

Reply:
they can't be private has they are called from outside. anyway I've added multi annotation and placed an ensure_one()